pkg/tasks/snapshot_helper.go

```diff
@@ -97,33 +102,79 @@ func (sh *SnapshotHelper) Run(versionHref string) error {
 	if err != nil {
 		return err
 	}
+	err = sh.payload.SaveSnapshotIdent(snap.UUID)
+	if err != nil {
+		return fmt.Errorf("unable to save snapshot ident: %w", err)
+	}
```

Comment:
It's a little confusing, but the SnapshotIdent is set previously and it's actually not the snapshot UUID, just a random UUID, because the snapshot object has not been created yet.

I see that you're setting it so that the snapshot can be deleted, which I think makes sense. To avoid conflating the two values, it probably makes sense to make a new "SnapshotUUID" field for the purpose of tracking snapshot creation?

Reply:
yeah, the `SnapshotIdent` confused me a little, but it wasn't used anywhere and I thought it should be the UUID so I "repurposed" it, but you are right it would be better to use a different separate field 😅 
changed it 👍🏼